AstarteDeviceSDKCSharp/Data/AstarteFailedMessageStorage.cs

```diff
@@ -28,48 +29,95 @@ namespace AstarteDeviceSDKCSharp.Data
 {
     public class AstarteFailedMessageStorage : IAstarteFailedMessageStorage
     {
-        private readonly AstarteDbContext _astarteDbContext;
-        private readonly AstarteDbContext _astarteDbContextDelete;
-        private readonly AstarteDbContext _astarteDbContextRead;
+        private readonly object _storedLock = new();
+        private readonly string _dbConnectionString;
 
+        SqliteConnection sqliteInsertConnection = new SqliteConnection();
+        SqliteConnection sqliteDeleteConnection = new SqliteConnection();
+        SqliteConnection sqliteReadConnection = new SqliteConnection();
         private static List<AstarteFailedMessageEntry> _astarteFailedMessageVolatile = new();
         readonly HashSet<ManagedMqttApplicationMessage> stored = new HashSet<ManagedMqttApplicationMessage>();
 
-
         public AstarteFailedMessageStorage(string persistencyDir)
         {
-            this._astarteDbContext = new AstarteDbContext(persistencyDir);
-            this._astarteDbContextDelete = new AstarteDbContext(persistencyDir);
-            this._astarteDbContextRead = new AstarteDbContext(persistencyDir);
+
+            _dbConnectionString = $"Filename = {persistencyDir}{Path.DirectorySeparatorChar}AstarteDeviceDb";
+
+            sqliteInsertConnection = new SqliteConnection(_dbConnectionString);
+            sqliteInsertConnection.Open();
+
+            sqliteDeleteConnection = new SqliteConnection(_dbConnectionString);
+            sqliteDeleteConnection.Open();
```

Comment:
Since you are locking on the insert and delete operation, keeping two separated connections seems excessive, so I think you can merge the two connection(InsertConnection and DeleteConnection) in one and save memory usage. What do you think?

Reply:
I merged that two connection, and tested it again. Seems to work fine.